modules/kafka/kafka-dest-worker.c

```diff
@@ -61,16 +62,38 @@ _publish_message(KafkaDestWorker *self, LogMessage *msg)
 {
   KafkaDestDriver *owner = (KafkaDestDriver *) self->super.owner;
   int block_flag = _is_poller_thread(self) ? 0 : RD_KAFKA_MSG_F_BLOCK;
+  rd_kafka_topic_t *topic;
+  //GString *topic_name_new = g_string_sized_new(128);
+  GString *topic_name_new = scratch_buffers_alloc();
```

Comment:
Don't we need to use a `Marker` here?

Reply:
Or another alternative would be a reusable GString in the worker, like topic_name_buffer.

Reply:
Can you clarify on the reusable GString part?

Reply:
Every worker could have a `GString *topic_name_buffer`, and the topic template could be resolved to there in each `_publish_message()` call. Every data stored in the `Worker` is safe and won't be accessed concurrently in the `publish_message()` function.

Reply:
So basically a global variable inside worker?

Reply:
Also, I wont be able to use scratch buffer in that case right?